src/utils/events/event-manager.js

```diff
@@ -0,0 +1,170 @@
+
+    this._onBasicInput = this._onBasicInput.bind(this);
+    this.manager = new Manager(element, {recognizers: RECOGNIZERS})
+      .on('hammer.input', this._onBasicInput);
+
+    this.aliasedEventHandlers = {};
+
+    // Handle events not handled by Hammer.js:
+    // - mouse wheel
+    // - pointer/touch/mouse move
+    this._onOtherEvent = this._onOtherEvent.bind(this);
+    this.wheelInput = new WheelInput(element, this._onOtherEvent);
+    this.moveInput = new MoveInput(element, this._onOtherEvent);
+
+    // Register all passed events.
+    const {events} = options;
+    if (events) {
+      this.on(events);
+    }
+  }
+
+  destroy() {
+    this.wheelInput.destroy();
+    this.moveInput.destroy();
+    this.manager.destroy();
+  }
+
+  /**
+   * Register an event handler function to be called on `event`.
+   * @param {string|Object} event   An event name (String) or map of event names to handlers.
+   * @param {Function} [handler]    The function to be called on `event`.
+   */
+  on(event, handler) {
+    if (typeof event === 'string') {
+      // Special handling for gestural events.
+      const recognizerEvent = EVENT_RECOGNIZER_MAP[event];
+      if (recognizerEvent) {
+        // Enable recognizer for this event.
+        this.manager.get(recognizerEvent).set({enable: true});
+
+        // Handle concurrent single and double tap registration as necessary.
+        this._reconcileSingleAndDoubleTap(recognizerEvent);
+
+        // Alias to a recognized gesture as necessary.
+        const eventAlias = GESTURE_EVENT_ALIASES[event];
+        if (eventAlias && !this.aliasedEventHandlers[event]) {
+          const aliasedEventHandler = this._aliasEventHandler(event);
+          this.manager.on(eventAlias, aliasedEventHandler);
+          // TODO: multiple handlers for the same aliased event will override one another.
+          // This should be an array of aliased handlers instead.
+          this.aliasedEventHandlers[event] = aliasedEventHandler;
+        }
+      }
+
+      // Register event handler.
+      this.manager.on(event, handler);
+    } else {
+      // If `event` is a map, call `on()` for each entry.
+      for (const eventName in event) {
+        this.on(eventName, event[eventName]);
+      }
+    }
+  }
+
+  /**
+   * Deregister a previously-registered event handler.
+   * @param {string|Object} event   An event name (String) or map of event names to handlers
+   * @param {Function} [handler]    The function to be called on `event`.
+   */
+  off(event, handler) {
+    if (typeof event === 'string') {
+      // Clean up aliased gesture handler as necessary.
+      const recognizerEvent = EVENT_RECOGNIZER_MAP[event];
+      if (recognizerEvent) {
+        const eventAlias = GESTURE_EVENT_ALIASES[event];
+        if (eventAlias && this.aliasedEventHandlers[event]) {
+          this.manager.off(eventAlias, this.aliasedEventHandlers[event]);
+          delete this.aliasedEventHandlers[event];
```

Comment:
Multiple `click` handlers will currently not work correctly as only one handler per alias can exist right now. See my TODO above ;)